AwsCryptographyPrimitives/runtimes/go/ImplementationFromDafny-go/AESEncryption/externs.go

```diff
@@ -0,0 +1,89 @@
+package AESEncryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"fmt"
+
+	"github.com/aws/aws-cryptographic-material-providers-library/primitives/AwsCryptographyPrimitivesTypes"
+	"github.com/dafny-lang/DafnyRuntimeGo/v4/dafny"
+	"github.com/dafny-lang/DafnyStandardLibGo/Wrappers"
+)
+
+var m_AESEncryption struct {
+	AES_GCM CompanionStruct_Default___
+}
+
+func (CompanionStruct_Default___) AESDecryptExtern(algo AwsCryptographyPrimitivesTypes.AES__GCM, key dafny.Sequence, cipherText dafny.Sequence, authTag dafny.Sequence, iv dafny.Sequence, aad dafny.Sequence) Wrappers.Result {
+	keyBytes := dafny.ToByteArray(key)
+	cipherTextBytes := dafny.ToByteArray(cipherText)
+	authTagBytes := dafny.ToByteArray(authTag)
+	ivBytes := dafny.ToByteArray(iv)
+	aadBytes := dafny.ToByteArray(aad)
+
+	if algo.Dtor_keyLength() != int32(len(keyBytes)) {
+		return Wrappers.Companion_Result_.Create_Failure_(AwsCryptographyPrimitivesTypes.Companion_Error_.Create_AwsCryptographicPrimitivesError_(dafny.SeqOfChars([]dafny.Char(fmt.Errorf("algorithm key length %d doesn't match actual key length %d ", algo.Dtor_keyLength(), len(keyBytes)).Error())...)))
```

Comment:
Copying from slack:

```
The actual implementation is barely ~70 lines long and has only two methods. We can break it but then it'll end up with duplicate code and each file with 50 lines
[1:37](https://amzn-aws.slack.com/archives/D04EVC03ZDG/p1733780241881429)
And it will break Lucas' script for what gets checked-in in the repo
1:38
I think the reason it's hard to read is due to dafny style Go
```

Also Go doesn't enforce any style - `go fmt` and `goimports` both pass this code. `golangci-lint` fails on dafny code. So I've manually tried to add line breaks wherever it made sense.